engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -45,13 +45,15 @@
     implements TypedRecordProcessor<ProcessInstanceModificationRecord> {
 
   private static final String ERROR_MESSAGE_PROCESS_INSTANCE_NOT_FOUND =
-      "Expected to modify process instance but no process instance found with key '%d'";
+      "Expected to modify instance of process but no process instance was found with key '%d'";
   private static final String ERROR_MESSAGE_TARGET_ELEMENT_NOT_FOUND =
-      "Expected to activate element but no element found in process '%s' for element id(s): '%s'";
+      "Expected to modify instance of process '%s' but it contained one or more activate instructions"
+          + " for elements that are not found within the process: '%s'";
```

Comment:
> `but it contained ...`

🔧 it didn't change, so contained should be in present tense -> `but it contains ...`

> `are not found within the process:`

🔧 can be reduced -> `cannot be found:`

> `but it contained one or more activate instructions for elements that are not found within the process:`

🔧 this reads as if the list at the ends refers to the activate instructions. It should be clear that these are element ids. Perhaps you could append `with element id` to this part of the message.

💭 If necessary, we could drop the part about activate instructions, but I do prefer that it's specific so a user can quickly find what's wrong, even when they sent a complex modification command.

Reply:
We should keep the part of it being an activate instruction, as it could also be a terminate instruction. Users can probably still figure out which of the 2 it is, but this makes it much more clear.

WDYT of `Expected to modify instance of process '%s' but it contains one or more activate instructions with an element that could not be found: '%s'`?
The way I read this we refer to the element ids, but I read it the same way in the old version as well 😄 I don't think it matters too much as an activate instruction has no significant identifier, except for the element id anyways.